proposals/0000-proposal-process.rst

```diff
@@ -0,0 +1,319 @@
+  light of a set of defined criteria (see below).
+
+* *Judging whether the proposal process should be applied.* There will no doubt
+  be proposals where it is unclear whether the burden of the proposal process is
+  necessary. The responsibility for determining the scope of the process lies
+  with the committee.
+
+* *Ensuring that the process is applied when necessary.* The committee will be
+  expected to monitor GHC contributions and ensure that changes which fall
+  within the scope of the proposal process indeed go through the process.
+
+Criteria for new features
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The committee is responsible for weighing a variety of factors when deciding
+whether to adopt a feature. These are all judgement calls.
+
+1. *The problem*. What exactly is the problem that the feature solves? Is
+   it an important problem, felt by many users, or is it very specialised?
+
+2. *The right solution; generality*. Elegance, economy, and generality are
+   important. Sometimes a problem has lots of solutions, none of which have that
+   "aha" feeling of "this is the Right way to solve this". A classic example is
+   that of records, for which we have had endless proposals (including many from
+   the GHC authors themselves) none of which felt "right", and none of which
+   made it into GHC.
+
+3. *Fit with the language*. If we just throw things into GHC willy-nilly, it
+   will become a large ball of incoherent and inconsistent mud. We strive to add
+   features that are consistent with the rest of the language.
+
+4. *Specification cost*. Does the benefit of the feature justify the extra
+   complexity in the language specification? Does the new feature interact
+   awkwardly with existing features, or does it enhance them? How easy is it for
+   users to understand the new feature?
+
+5. *Implementation cost*. How complex is the implementation likely to be? Is it
+   highly localised, or does it require pervasive changes? As an author and
+   contributor, it may look like you are giving GHC free work, but this
+   viewpoint doesn't align with the reality of a large project with a timescale
+   in decades. Writing code is cheap; maintaining it is expensive.
+
+
+Drawbacks
+---------
+
+Of course, group decision-making processes are difficult to manage and tools
+will only bring you so far. While the Rust scheme does seem to function more
+smoothly than our current system, it is not free of issues (as most recently
+discussed by `Aaron Turon
+<https://aturon.github.io/blog/2016/07/05/rfc-refinement/>`_). These issues will
+likely apply to the process proposed here as well. In particular,
+
+* GitHub discussions in particular don't scale terribly well; the lack of
+  hierarchical threading means that long threads can become difficult to follow;
+  experience suggests that Phabricator seems to be a bit better but long
+  technical discussions are a challenge for nearly any tool.
```

Comment:
No matter the tool, we should be mindful of the risk that limitations will prompt branching off discussion in the form of IRC chats, skype calls, etc. I'm not sure that a _mechanism_ is appropriate to address this, but proposal committee members should try to report back the salient points of such conversations in the originating thread. One of the ways I've seen people feel excluded by the current system is when this doesn't happen.